### PR TITLE
L5 m0/minesweeper

### PR DESCRIPTION
**Pull request by dencee**

### CODE REVIEW

File: Level5-Module0/src/_02_Intro_To_Streams/IntroToStreams.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * 
+ * Certain stream methods operate on the collection and return a stream with
+ * the changes, similar to how string methods return a new string. Here is an
+ * example that counts the total number of characters of strings larger than 4:
+ *   int totalCharacters = Arrays.stream(words)
+ *                          .filter((word) -> word.length() > 4)
+ *                          .map((word) -> word.length())
+ *                          .reduce(0, (int1, int2) -> int1 + int2);
+ * filter() and map() both return a stream so the methods can be called right
+ * afterwards. reduce() returns an int  and closes the stream so it must be
```
Comment (mjfre) [inline Level5-Module0/src/_02_Intro_To_Streams/IntroToStreams.java:41]: Could you explain what the reduce does a little more?  I'm not sure I really understand it.

File: Level5-Module0/src/_05_Minesweeper/Minesweeper.java [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    /*
+     * A cell was clicked and this method needs to:
+     * 1. Only do the following steps if the cell is NOT a mine.
+     * 2. Mark the cell as revealed.
+     * 3. If there are no mines around this cell (cell.minesAround == 0),
+     *    call the getNeighbors() method to return a list of all the
+     *    surrounding Cell objects.
+     * 4. Use a stream to reveal any of the neighboring cells in the list.
+     *    The filter() and/or forEach() methods may be useful.
+     * 5. Use recursion to check any neighbors of those neighbors that may also
```
Comment (mjfre) [inline Level5-Module0/src/_05_Minesweeper/Minesweeper.java:91]: Some further explanation might be helpful here.

**Review by mjfre — APPROVED** [review]
Looks really good.  I feel like I understand streams a little better after reading through it.

File: Level5-Module0/src/_02_Intro_To_Streams/IntroToStreams.java (hunk 1 above)
Comment (dencee) [reply Level5-Module0/src/_02_Intro_To_Streams/IntroToStreams.java]: I updated the explanation. It combines all the elements in the stream to a single element (reducing).
.reduce(initialValue, (accumulator, nextValue) -> accumlator + nextValue;
The code above does a sum on all the stream elements and returns the final value in the accumulator.

File: Level5-Module0/src/_05_Minesweeper/Minesweeper.java (hunk 2 above)
Comment (dencee) [reply Level5-Module0/src/_05_Minesweeper/Minesweeper.java]: done!

**Pull request merged**